src/CosmosClient.ts

```diff
@@ -51,7 +52,11 @@ export class CosmosClient {
    */
 
   private clientContext: ClientContext;
-  constructor(private options: CosmosClientOptions) {
+  constructor(options: string | CosmosClientOptions) {
```

Comment:
I know it's annoying, but can we do this with an overload instead?

```typescript
constructor(connectionString: string)
constructor(options: CosmosClientOptions)
constructor(optionsOrConnectionString: string | CosmosClientOptions)
```

Even if you don't want to overload it, you should at least use the optionsOrConnectionString name (but I like overloading :))

Reply:
Also, need to update the JSdocs

Reply:
Done. Added overload and docs